pkg/git/service.go

```diff
@@ -102,7 +102,7 @@ func (s *Service) CloneRepository(project *project.Project, auth *http.BasicAuth
 func (s *Service) CloneRepositoryCmd(p *project.Project, auth *http.BasicAuth) []string {
```

Comment:
Changes you made here look good, but you need to apply that to the `CloneRepository` method as well.

P.S. I think you can remove the `shouldCheckoutSha` and `shouldCloneBranch` methods now.